test/replication-luatest/gh_7590_replicaset_assert_test.lua

```diff
@@ -0,0 +1,91 @@
+local t = require('luatest')
+local server = require('luatest.server')
+local cluster = require('luatest.replica_set')
+
+local g = t.group('gh-7590')
+
+g.before_all(function(g)
+    g.cluster = cluster:new({})
+    local replication_cfg = {
+        server.build_listen_uri('r1'),
+        server.build_listen_uri('r2'),
+        server.build_listen_uri('r3'),
+    }
+
+    g.r1 = g.cluster:build_and_add_server({alias = 'r1', box_cfg = {
+        -- We need to specify uuid explicitly, so that rb tree traversal
+        -- wouldn't take errored applier (applier from r2 to r3) first.
+        instance_uuid = '00000000-0000-0000-0000-000000000001',
+        replication = replication_cfg,
```

Comment:
Let's also set `replication_timeout = 0.1`, like it's done in other tests.
This speeds them up quite a bit. For example, your test would run for ~1 second instead of ~3 on my machine.

Reply:
Fixed